subsys/llext/llext_mem.c

```diff
@@ -76,15 +76,23 @@ static int llext_copy_region(struct llext_loader *ldr, struct llext *ext,
 			/* Directly use data from the ELF buffer if peek() is supported */
 			ext->mem[mem_idx] = llext_peek(ldr, region->sh_offset);
 			if (ext->mem[mem_idx]) {
-				llext_init_mem_part(ext, mem_idx, (uintptr_t)ext->mem[mem_idx],
-						    region_alloc);
-				ext->mem_on_heap[mem_idx] = false;
-				return 0;
+				if (!IS_ALIGNED(ext->mem[mem_idx], region_align) &&
+				    !(ldr_parm && ldr_parm->pre_located)) {
```

Comment:
you could revert the condition and remove the `else` because now the `if` would contain a `return`